src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -279,6 +279,7 @@ private CookieOptions BuildCookieOptions()
             var context = new CookieSigningOutContext(
                 Context,
                 Options,
+                new AuthenticationProperties(signOutContext.Properties),
```

Comment:
No, it's not internal, it's private and set from the constructor :trollface: 
Hence my remark.

> We don't care if anyone changes it, we won't flow it.

Not sure why you want to let the user think he's allowed to replace the instance, then.